### ISSUE TITLE
Add support for multi-threaded test execution

### ISSUE THREAD

**Issue by cschreib** [cschreib/snatch#33]
The framework is written to be thread-safe, but there is currently no way to run tests in parallel. This should be easy to add, but perhaps not without using heap memory. In which case, this will have to be guarded by macros, so it can be disabled if necessary.

**Comment by AdelKS** [comment]
Another possibility is to implement what gtest calls "sharding" which is a multi processing approach. You tell the test excitable when running its number and how many you are going to run, and it will run a specific subset of the tests. 

I am writing it here because it didn't see that it's supported in the readme.

**Comment by cschreib** [comment]
Hi and thanks for the tip! You are correct, this is not currently supported. I know *Catch2* supports this too. Do you know how the different processes get spawned? I.e., is this meant to be supported by an external scheduler / test runner? And how do you know how it meshes with test filtering?

**Comment by AdelKS** [comment]
> Do you know how the different processes get spawned? I.e., is this meant to be supported by an external scheduler / test runner?

Yes, the user has to run the test executable `N` times, and tells each executable their index `i` and how many subdivisions, `N` he wants on the tests.

> And do you know how it meshes with test filtering?

I do not unfortunately. Although I think it's best to filter first and then subdivide the subset that remains, so it probably does that.
